langkit/libmanage.py

```diff
@@ -26,62 +29,92 @@
 
 if TYPE_CHECKING:
     from langkit.compile_context import CompileCtx
+    from types import TracebackType
 
 
 class Directories:
     """
     Helper class used to get various path in source/build/install trees.
     """
 
-    def __init__(self, lang_source_dir=None, build_dir=None, install_dir=None):
+    def __init__(self,
+                 lang_source_dir: str,
+                 build_dir: Opt[str] = None,
+                 install_dir: Opt[str] = None):
+
         self.root_lang_source_dir = lang_source_dir
         self.root_build_dir = build_dir
         self.root_install_dir = install_dir
 
-    def set_build_dir(self, build_dir):
+    def set_build_dir(self, build_dir: str) -> None:
         self.root_build_dir = build_dir
 
-    def set_install_dir(self, install_dir):
+    def set_install_dir(self, install_dir: str) -> None:
         self.root_install_dir = path.abspath(install_dir)
 
-    def lang_source_dir(self, *args):
+    def lang_source_dir(self, *args: str) -> str:
         return path.join(self.root_lang_source_dir, *args)
 
-    def build_dir(self, *args):
+    def build_dir(self, *args: str) -> str:
+        assert self.root_build_dir is not None
         return self.lang_source_dir(
             self.root_lang_source_dir, self.root_build_dir, *args
         )
 
-    def install_dir(self, *args):
+    def install_dir(self, *args: str) -> str:
+        assert self.root_install_dir is not None
         return path.join(self.root_install_dir, *args)
 
 
 class EnableWarningAction(argparse.Action):
-    def __call__(self, parser, namespace, values, option_string=None):
+    def __call__(self,
+                 parser: argparse.ArgumentParser,
+                 namespace: argparse.Namespace,
+                 values: Union[Text, Sequence[Any], None],
+                 option_string: Opt[Text] = None) -> None:
         namespace.enabled_warnings.enable(values)
 
 
 class DisableWarningAction(argparse.Action):
-    def __call__(self, parser, namespace, values, option_string=None):
+    def __call__(self,
+                 parser: argparse.ArgumentParser,
+                 namespace: argparse.Namespace,
+                 values: Union[Text, Sequence[Any], None],
+                 option_string: Opt[Text] = None) -> None:
+
         namespace.enabled_warnings.disable(values)
 
 
 class ManageScript:
 
     BUILD_MODES = ('dev', 'prod')
 
+    build_mode: str
+    """
+    Build mode for the project.
+    """
+
     ENABLE_BUILD_WARNINGS_DEFAULT = False
     """
     Whether warnings to build the generated library are enabled by default.
     """
 
+    enable_build_warnings: bool
+    """
+    Whether to enable build warnings.
+    """
+
     # The create_context method will create the context and set it
     # only right before executing commands.
     context: 'CompileCtx'
     """
     Langkit compilation context.
     """
 
+    # This will be set in the run method, when we have parsed arguments
```

Comment:
`run` method
(IIRC we have agreed to quote references to code)

Reply:
We have agreed to quote references to code in docstrings, where this will actually have an impact on generated documentation. I don't think we ever agreed to do that in comments, where it doesn't make a whole lot of sense IMO. I'm OK to discuss it but that's a different issue in my view.

Reply:
Ok, let’s discuss :) I did not realize we agreed to do that only for doc generation, and I’m fine with this refinment. However in other places, I would advise to surround with double quotes or backquotes as soon as it’s not obvious that the name is a code entity (mostly: when there’s no underscore in the name, so apply to `run` here), as it helps reading (at least to me). WDYT?

Reply:
I'm fine with it I guess, which syntax do you want to use ?

Reply:
On one hand I find double quotes easier to type, but on the other hand it’s inconsistent with the backquotes we use everywhere else (i.e. in docstrings for for Markdown). I’m fine with both options as long as we stick to the rule. Do you have a preference, in the light of this?

Reply:
I dunno :( Also the doc style is *double* backquotes, which is even more tedious to type

Reply:
Let's go for ""